pkg/cache/context.go

```diff
@@ -606,6 +606,17 @@ func (ctx *Context) PublishEvents(eventRecords []*si.EventRecord) {
 						zap.String("taskID", taskID),
 						zap.String("event", record.String()))
 				}
+			case si.EventRecord_NODE:
+				nodeID := record.ObjectID
+				node, err := ctx.apiProvider.GetAPIs().NodeInformer.Lister().Get(nodeID)
```

Comment:
this will cause the shim to query API server once the `PublishEvents` is called. this could be adding a big overhead to api-server. Instead of doing this, we should leverage the cached nodes in `pkg/cache/external/scheduler_cache.go`. we should be able to get the node object by `ctx.schedulerCache.GetNode(node).Node()`

Reply:
Agreed, fixed.